apollo-composition/src/lib.rs

```diff
@@ -355,3 +356,58 @@ fn satisfiability_result_into_issues(
         Err(issue) => Either::Right(once(issue)),
     }
 }
+
+impl From<BuildError> for Issue {
+    fn from(error: BuildError) -> Issue {
+        Issue {
+            code: error.code.expect("BuildError.code unexpectedly missing"),
```

Comment:
`.expect` is for things that never happen. Does this never happen? If it doesn't, should BuildError be changed to not use `Option`s?

Reply:
Since all these things come from TypeScript, and we have to support arbitrarily old versions, we can't be sure they'll never be null. `unwrap_or_default()` here would be better here. Message should probably have a default string, like "Unknown error" or something. Locations should be able to `into_iter()` on the option to end up with an empty vec if it was null.